apps/ledger-live-desktop/src/renderer/screens/exchange/Swap2/Form/FeesDrawerLiveApp/index.tsx

```diff
@@ -40,29 +40,41 @@ export default function FeesDrawerLiveApp({
 
   const handleSetTransaction = useCallback(
     (transaction: Transaction) => {
-      setTransactionState(transaction);
-      setTransaction(transaction);
+      const account = mainAccount.type === "TokenAccount" ? parentAccount : mainAccount;
       bridge
-        .getTransactionStatus(
-          mainAccount.type === "TokenAccount" ? parentAccount : mainAccount,
-          transaction,
+        .prepareTransaction(account, transaction)
+        .then(preparedTransaction =>
+          bridge.getTransactionStatus(account, preparedTransaction).then(status => {
+            setTransactionStatus(status);
+            setTransactionState(preparedTransaction);
+            setTransaction(preparedTransaction);
+          }),
         )
-        .then(setTransactionStatus);
+        .catch(error => {
+          console.error("Error preparing transaction:", error);
```

Comment:
Do we need to throw any errors here for display on the frontend ?

Reply:
agree with displaying the error but no components/design (all other errors now are inside coin-xxx)
It will be the work of another PR